agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/perfcounter/OshiPerformanceCounter.java

```diff
@@ -55,10 +57,22 @@ public String getId() {
   public void report(TelemetryClient telemetryClient) {
     if (processInfo == null || processor == null) {
       // lazy initializing these because they add to slowness during startup
-      SystemInfo systemInfo = new SystemInfo();
-      OperatingSystem osInfo = systemInfo.getOperatingSystem();
-      processInfo = osInfo.getProcess(osInfo.getProcessId());
-      processor = systemInfo.getHardware().getProcessor();
+      try {
+        SystemInfo systemInfo = new SystemInfo();
+        OperatingSystem osInfo = systemInfo.getOperatingSystem();
+        processInfo = osInfo.getProcess(osInfo.getProcessId());
+        processor = systemInfo.getHardware().getProcessor();
+      } catch (Error ex) {
+        // e.g. icm 253155448: NoClassDefFoundError
+        // e.g. icm 276640835: ExceptionInInitializerError
+        hasError.set(true);
+        return;
+      }
+    }
+
+    // stop collecting oshi perf counters when initialization fails.
+    if (hasError.get()) {
+      return;
     }
```

Comment:
should this go at the beginning of the method?

Reply:
it doesn't make much diff.. except it avoids checking if it has been init or not..  i can move it.

Reply:
it will never get here tho?

Reply:
1st init -> has error then return, 
2nd processInfo is not null -> hit line 74

Reply:
`processorInfo` will still be null on 2nd, because exception was thrown before it could be set